### PR TITLE
Th2 1836 workflow changes

### PR DESCRIPTION
**Pull request by Andrei-Shulika**
Adding license check to main workflows with the ability to fail

### CODE REVIEW

File: .github/workflows/license_check.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
       runsOn: ${{ inputs.runsOn }}
 
   check_licenses:
+    continue-on-error: true
```
Comment (Nikita-Smirnov-Exactpro) [inline .github/workflows/license_check.yml:19]: suggestion: Let's add option with true value as default

File: .github/workflows/compound-java-dev.yml [hunk 2]
```diff
@@ -133,3 +133,6 @@ jobs:
       javaVendor: ${{ inputs.javaVendor }}
     secrets:
       nvd-api-key: ${{ secrets.nvd-api-key }}
+
+  license-check:
+    uses: th2-net/.github/.github/workflows/license_check.yml@main
```
Comment (Nikita-Smirnov-Exactpro) [inline .github/workflows/compound-java-dev.yml:138]:
```suggestion
    if: inputs.scanner-enabled
    uses: th2-net/.github/.github/workflows/license_check.yml@main
```

File: .github/workflows/license_check.yml (hunk 1 above)
Comment (Andrei-Shulika) [reply .github/workflows/license_check.yml:19]: agreed. done

File: .github/workflows/compound-java-dev.yml (hunk 2 above)
Comment (Andrei-Shulika) [reply .github/workflows/compound-java-dev.yml:139]: If condition added to all files where license_check called

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**